src/index.js

```diff
@@ -102,39 +132,47 @@ class Dropzone extends React.Component {
 
   onDrop(e) {
     const { onDrop, onDropAccepted, onDropRejected, multiple, disablePreview } = this.props;
-    const fileList = getDataTransferItems(e, multiple);
     const acceptedFiles = [];
     const rejectedFiles = [];
 
     // Stop default browser behavior
     e.preventDefault();
 
-    // Reset the counter along with the drag on a drop.
-    this.enterCounter = 0;
+    // Reset the target list along with the drag on a drop.
+    this.dragTargets = [];
     this.isFileDialogActive = false;
 
-    fileList.forEach((file) => {
-      if (!disablePreview) {
-        file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
-      }
+    // Don't fire callbacks if a child Dropzone component has already handled this event
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.__dropzoneChildHandledThis) {
```

Comment:
Awesome!

> I think it might be surprising for someone who tries nesting to find that the parent still receives a drop event

TBH I would be surprised if it wouldn't receive and fire the event :)